Hal porting

fpistm opened the pull request:
Ciao Francesco,

Here the full HAL porting.
Audio is ~~missing but will be available soon.~~ available
DFU is ~~not~~ implemented. (thanks @cparata)

Tested with Arduino 1.8.2.

You will need to install CMSIS 4.5.0 package.

fpistm committed:
Clean up files

Remove CMSIS sources, package CMSIS 4.5.0 from arduino
will be uses as dependency.
Remove static library.
Remove FatFs.

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Update to the STM32CubeF4 V1.16.0

HAL and CMSIS device.

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Update core based on HAL

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Add HAL wrap

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Add STM32 core

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Update Otto variant

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Update libraries

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Update boards.txt and platform.txt

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Add Audio library

2 examples
- Player
- recorder

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Fix macro

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
micros() is going backwards

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Wrong definition for pins

Moreover it seems there is a pin numbering misalignement between the schematic
and the printed one on the connector.

Connector start at 22 while shcematics start a 23 (no 22 on the schematics)

Possible enhancement:
D77 could be removed
and
#define LCD_BL                  D77
could be set to D22

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

cparata committed:
Add Wire1 and Wire2 instances

Signed-off-by: Carlo.Parata <carlo.parata@st.com>

cparata committed:
Add DFU

Signed-off-by: Carlo.Parata <carlo.parata@st.com>

cparata committed:
Fix for USB serial issue

Signed-off-by: Carlo.Parata <carlo.parata@st.com>

fpistm committed:
Remove unused .configured

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

alfran commented:
I found some issue:
1) PWM not works
2) SPI connected to ICSP connector not works
3) I2C and I2C1 not works
4) Recording using The microphone the Audio "jump" (some part of audio data is lost)

fpistm commented:
There was an issue when I add commit from Carlo, that's why PWM is not working. Now it is corrected.
For Audio and I2C, I do not see any issue.
SPI have to be corrected.

Commit:
Fix SPI constructors and SS pin usage modified

Signed-off-by: fpr <fabien.perroquin@wi6labs.com>

Commit:
In uart_debug_write() check transmit status to be sure data has been sent

Signed-off-by: fpr <fabien.perroquin@wi6labs.com>

fpistm committed:
Fix PWM frequency

Get the correct timer freq depending on timer clock source.

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Fix SPI clock freq definition

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

Commit:
Global variables removed (more secure). Exit init if pins are NC.

Signed-off-by: fpr <fabien.perroquin@wi6labs.com>

fpistm committed:
Update SPI pins definition and add instance

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Fix build issue

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
TIMER_TONE freq is twice frequency

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Use spi_transfer instead of spi_send

To be inline with https://www.arduino.cc/en/Reference/SPITransfer
"In case of buffer transfers the received data is stored in the buffer
in-place (the old data is replaced with the data received)."

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

cparata committed:
Fix SPI clock polarity issue

PULLDOWN or PULLUP the SCK pin according the polarity used.

Fix #24

Signed-off-by: Carlo.Parata <carlo.parata@st.com>

cparata committed:
Fix DFU mode issue

Issue was due to the fact that sometimes the code remained blocked
inside the timer callback after the closure of the USB serial port.
In order to avoid that, some checks inside the timer callbeck and
the CDC_Flush API have been added.

Improving the calculation of the period and prescaler of the timer.

Signed-off-by: Carlo.Parata <carlo.parata@st.com>

fpistm committed:
ARR register have to be set at Period-1.

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
Review analog pins numbering and pin functions

analogRead(A0) == analogRead(0)
analogWrite(A0) != analogWrite(0) == analogWrite(D0)
All pins available in PinMap_ADC are available with Ax alias

Move pin functions to pins_arduino.c to avoid duplicated code.
Pin functions naming alignement.

Move init() function to pins_arduino.c to avoid duplicated code
and define it as weak.

Minor clean of unused code.
Remove usb flags for variants w/o usb.
Serialx under switch.Disable by default

Now, variant files only contain source which differ for each variants.

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

alfran commented:
@cparata, @fpistm: Thank you so much guys! was a pleasure!
